src/EditorFeatures/VisualBasicTest/ConvertForToForEach/ConvertForToForEachTests.vb

```diff
@@ -0,0 +1,533 @@
+    sub Test(array as string())
+        dim i as integer
+        [||]For i = 0 to array.Length - 1 step 2
+            Console.WriteLine(array(i))
+        next
+    end sub
+end class")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertForToForEach)>
+        Public Async Function TestMissingWithIncorrectCondition1() As Task
+            Await TestMissingInRegularAndScriptAsync(
+"imports System
+
+class C
+    sub Test(array as string())
+        [||]For i = 0 to array.Length
+        {
+            Console.WriteLine(array(i))
+        next
+    end sub
+end class")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertForToForEach)>
+        Public Async Function TestMissingWithIncorrectCondition2() As Task
+            Await TestMissingInRegularAndScriptAsync(
+"imports System
+
+class C
+    sub Test(array as string())
+        [||]For i = 0 to GetLength(array) - 1 
+        {
+            Console.WriteLine(array(i))
+        next
+    end sub
+end class")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertForToForEach)>
+        Public Async Function TestMissingWithIncorrectCondition3() As Task
+            Await TestMissingInRegularAndScriptAsync(
+"imports System
+
+class C
+    sub Test(array as string())
+        [||]For i = 0 to array.Length - 2
+        {
+            Console.WriteLine(array(i))
+        next
+    end sub
+end class")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertForToForEach)>
+        Public Async Function TestNotStartingAtZero() As Task
+            Await TestMissingInRegularAndScriptAsync(
+"imports System
+
+class C
+    sub Test(array as string())
+        [||]For i = 1 to array.Length - 1
+        {
+            Console.WriteLine(array(i))
+        next
+    end sub
+end class")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertForToForEach)>
+        Public Async Function TestList1() As Task
+            Await TestInRegularAndScript1Async(
+"imports System
+imports System.Collections.Generic
+
+class C
+    sub Test(list as IList(of string))
+        [||]For i = 0 to list.Count - 1
+            Console.WriteLine(list(i))
+        next
+    end sub
+end class",
+"imports System
+imports System.Collections.Generic
+
+class C
+    sub Test(list as IList(of string))
+        For Each {|Rename:v|} In list
+            Console.WriteLine(v)
+        next
+    end sub
+end class")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertForToForEach)>
+        Public Async Function TestChooseNameFromDeclarationStatement() As Task
+            Await TestInRegularAndScript1Async(
+"imports System
+imports System.Collections.Generic
+
+class C
+    sub Test(list as IList(of string))
+        [||]For i = 0 to list.Count - 1
+            dim val = list(i)
```

Comment:
I think I didn't see it from C# as well. so if there is 2 dim val = list(i) and dim val2 = list(i), I guess the first one get picked up and second one becomes dim val2 = val?

Reply:
yes.  only the first statement is checked to see if it should be lifted into the variable in the loop.